app/api/csv/arrangeThesauri.ts

```diff
@@ -0,0 +1,180 @@
+import { ImportFile } from 'api/csv/importFile';
+import thesauri from 'api/thesauri';
+import { propertyTypes } from 'shared/propertyTypes';
+import { PropertySchema } from 'shared/types/commonTypes';
+import { TemplateSchema } from 'shared/types/templateType';
+import { ThesaurusSchema } from 'shared/types/thesaurusType';
+
+import csv, { CSVRow } from './csv';
+import { splitMultiselectLabels } from './typeParsers/multiselect';
+import { normalizeThesaurusLabel } from './typeParsers/select';
+
+const filterJSObject = (input: { [k: string]: any }, keys: string[]): { [k: string]: any } => {
+  const result: { [k: string]: any } = {};
+  keys.forEach(k => {
+    if (input.hasOwnProperty(k)) {
+      result[k] = input[k];
+    }
+  });
+  return result;
+};
+
+class ArrangeThesauriError extends Error {
+  source: Error;
+  row: CSVRow;
+  index: number;
+
+  constructor(source: Error, row: CSVRow, index: number) {
+    super(source.message);
+    this.source = source;
+    this.row = row;
+    this.index = index;
+  }
+}
+
+const separateSelectAndMultiselectThesauri = (
+  thesauriRelatedProperties: PropertySchema[] | undefined,
+  languages?: string[]
+): [{ [k: string]: string }, { [k: string]: string }] => {
+  const nameToThesauriIdSelects: { [k: string]: string } = {};
+  const nameToThesauriIdMultiselects: { [k: string]: string } = {};
+
+  thesauriRelatedProperties?.forEach(p => {
+    if (p.content && p.type) {
+      const thesarusID = p.content.toString();
+      if (p.type === propertyTypes.select) {
+        nameToThesauriIdSelects[p.name] = thesarusID;
+        languages?.forEach(suffix => {
+          nameToThesauriIdSelects[`${p.name}__${suffix}`] = thesarusID;
+        });
+      } else if (p.type === propertyTypes.multiselect) {
+        nameToThesauriIdMultiselects[p.name] = thesarusID;
+        languages?.forEach(suffix => {
+          nameToThesauriIdMultiselects[`${p.name}__${suffix}`] = thesarusID;
+        });
+      }
+    }
+  });
+
+  return [nameToThesauriIdSelects, nameToThesauriIdMultiselects];
+};
+
+type ThesauriValueData = {
+  thesauriIdToExistingValues: Map<string, Set<string>>;
+  thesauriIdToNewValues: Map<string, Set<string>>;
+  thesauriIdToNormalizedNewValues: Map<string, Set<string>>;
+};
+
+const setupIdValueMaps = (allRelatedThesauri: ThesaurusSchema[]): ThesauriValueData => {
+  const thesauriIdToExistingValues = new Map();
+  const thesauriIdToNewValues = new Map();
+  const thesauriIdToNormalizedNewValues = new Map();
+
+  allRelatedThesauri.forEach(t => {
+    if (t._id) {
```

Comment:
@LaszloKecskes same as a previous comment, thesauris.get already ensures the existence of _id, you just need to type the parameter properly with something like WithId<ThesaurusSchema>[]

Reply:
Done.